src/analysis/history_utils.hpp

```diff
@@ -31,26 +32,30 @@ using namespace Geometry;
 
 namespace History {
 
-template <typename Pack, typename Geometry>
-KOKKOS_INLINE_FUNCTION Real CalcMassFlux(Pack &pack, Geometry &geom, const int prho,
-                                         const int pvel_lo, const int pvel_hi,
-                                         const int b, const int k, const int j,
-                                         const int i) {
+template <typename Geometry>
+KOKKOS_INLINE_FUNCTION Real CalcMassFlux(MeshData<Real> *md, Geometry &geom, const int b,
```

Comment:
This is not going to work on device. You still need to pass in the pack. But you probably don't need to pass in `prho`, `pvel_lo` or `pvel_hi` anymore, as you can simply use the types inside the pack.